themes/bootstrap3/templates/channels/channelList.phtml

```diff
@@ -26,8 +26,8 @@
 <?php if (empty($token)): ?>
   <form action="<?=$this->url('channels-search')?>" class="channel-search form-inline">
     <?=$this->transEsc('channel_searchbox_label')?>
-    <input type="text" name="lookfor" class="form-control" value="<?=$this->escapeHtmlAttr($this->lookfor) ?>">
-    <input type="submit" value="<?=$this->transEscAttr('Submit')?>" class="btn btn-default">
+    <input type="text" name="lookfor" class="form-control" value="<?=$this->escapeHtmlAttr($this->lookfor) ?>" aria-label="<?=$this->transEscAttr('search_terms') ?>">
```

Comment:
Would it be a better enhancement to use `placeholder` instead of `aria-label`? This would add a visual clue and a screen reader hint.

https://russmaxdesign.github.io/accessible-forms/accessible-name-input05.html

Reply:
I think `placeholder` is not an equivalent alternative to `aria-label`, 
["placeholder text is not broadly supported across assistive technologies and not displayed in older web browsers" ](https://www.w3.org/WAI/tutorials/forms/instructions/)
But it seems there are different views on `placeholder` and accessebility, do you have more information about that @ckaz ?

Reply:
I'd also like a third opinion. I'm disappointed to see that W3 haven't updated that page in three years! The page I linked led me to believe that placeholder was supported by screen readers [and browsers](https://caniuse.com/css-placeholder) but I don't have a lot of hands on experience, nor a setup that would let me test options myself.

Reply:
> I think `placeholder` is not an equivalent alternative to `aria-label`,
> ["placeholder text is not broadly supported across assistive technologies and not displayed in older web browsers" ](https://www.w3.org/WAI/tutorials/forms/instructions/)
> But it seems there are different views on `placeholder` and accessebility, do you have more information about that @ckaz ?

I'd stick with labels/aria-labels -- placeholders could be used as an option for the libraries who want them. 

Apart from what you both wrote, another case against placeholders is that placeholders are for people who _get_ them, only. To those who don't, the form may look as if it had already been filled in, which may cause confusion, which may cause distrust in the site/product etc. etc. ... In addition, if there are _required_ fields in the form, reviewing errors may also become more difficult if placeholders are there. They pose an UX issue while improving usability for some users at the same time -- in my opinion at least.
Cheers!

Reply:
When I add a `placeholder` and start the Screenreader **Orca** with **Chrome** it is just reading the aria-label but not the placeholder. 
With **Firefox** and **Orca** it is reading both, aria-label and placeholder. 
It is just one screenreader, but for Orca it seems that aria-label works better and more reliable across different browsers.

Reply:
If placeholder is being completely ignored in some situations, I agree that we probably need to stick with aria-label. What do you think, @crhallberg? Are any further adjustments needed, or should we merge this one?